source/libs/cache.ts

```diff
@@ -45,30 +45,35 @@ export function set<TValue extends any = any>(key: string, value: TValue, expira
 }
 
 /* Accept messages in background page */
-if (!browser.runtime.getBackgroundPage) {
```

Comment:
I think the change required here is:

```js
if (location.pathname === '/_generated_background_page.html') {
```

because `browser.runtime.getBackgroundPage` was only truthy in the background page and this condition makes it exactly clear of what we're expecting.